### ISSUE TITLE
[function-documentation] Add rule for documenting class methods

### ISSUE THREAD

**Issue by rwdj** [eslint/eslint#12020]
# function-documentation

## Behavior

Style enforcement to require a commented line before every class method member.
Potentially with a default to exempt no parameter constructors.

## Category

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

## Examples

```js
class ExampleOne {
  // some comment
  constructor() {} // => no warning

  // some comment
  method1() { ... } // => no warning
 
  /**
   * Some comment
   */
  method3() { ... } // => no warning

  method2() { ... } // => warning
}

class ExampleTwo {
  constructor() {} // => no warning
}

class ExamplThree {
  constructor( ... ) {} // => no warning
}

class ExampleFour {
  constructor() { ... } // => warning
}

class ExampleFive {
  constructor( ... ) { ... } // => warning
}
```

And while we're at it, another rule for class-documentation would be similarly helpful.

## Why ESLint

"ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code." Sounds like this ECMAScript style choice fits.

Some projects are large, modulated and recompiled (compressed). These comments will be stripped away after compilation, but the source code is more maintainable if we enable a rule to be forced to at least comment every method.

I'm not asking for to use a specific style of comment (i.e. JDoc). So long as the previous line is some form of comment, the minimal purpose is fulfilled. 

## Willing to submit a pull request

Sure. Happily. But I'd have to I sort out how ESlint rules are made, so I might need a bit of time. Especially since I don't know how they work with old ECMAScript.

**Comment by rwdj** [comment]
A little disclaimer: I've never actually used pure ESMAScript. I've always used other forms. Especially typescript. So, let me know if there's style issues with this rule in ESMAScript. From what I know, this rule belongs in ESMAScript before typescript.

**Comment by platinumazure** [comment]
Hi @rwdj, thanks for the issue.

I'm personally not sure about adding this to core. We already have a lot of rules, so our standard for new rules to be added to core is very high. I don't see a strong need to add comment-based rules to core.

You are welcome to create a plugin or add your rule to an existing plugin if the team decides this is not a good fit for ESLint core.

**Comment by kaicataldo** [comment]
Agreed that this feels like a better fit for a plugin than core.
